File: epistimi-frontend/src/dto/noticeboard-post.ts
import { UserResponse } from './user';

export interface NoticeboardPostResponse {
  id: string;
  author: UserResponse;
  createdDate: Date;
  updatedDate: Date;
  title: string;
  content: string;
}

export interface NoticeboardPostResponse {
  posts: NoticeboardPostResponse[];
}

File: epistimi-backend/src/main/kotlin/pl/edu/wat/wcy/epistimi/noticeboard/NoticeboardPost.kt
package pl.edu.wat.wcy.epistimi.noticeboard

import pl.edu.wat.wcy.epistimi.organization.Organization
import pl.edu.wat.wcy.epistimi.user.User
import java.time.LocalDateTime
import java.util.UUID

data class NoticeboardPost(
    val id: NoticeboardPostId? = null,
    val organization: Organization,
    val author: User,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
    val title: String,
    val content: String,
)

@JvmInline
value class NoticeboardPostId(
    val value: UUID,
)
